linera-core/src/worker.rs

```diff
@@ -743,28 +743,22 @@ where
     ) -> Result<(), WorkerError> {
         let required_locations = block.bytecode_locations();
         // Find all certificates containing bytecode used when executing this block.
-        let mut required_hashes: HashSet<_> = required_locations
-            .keys()
-            .map(|bytecode_location| bytecode_location.certificate_hash)
-            .collect();
+        let mut required_locations_left: HashMap<CryptoHash, &BytecodeLocation> =
```

Comment:
Optional nit: I think I'd call this `missing_locations` instead :thinking:

Reply:
I think that can be confusing because the `locations` variable below is the actual `missing_locations` one 🤔 and this one, since we remove things from it, is the required ones that are still left to check. So I might change the one below for `missing_locations` and keep this one as is 🤔